File: formacao_cientista/2. distribuicoes/dist_binomial.R

# dbinom -> probabilidade individual
num_sucessos = 3
num_experimentos = 5
prob = 0.5

psucessos = dbinom(num_sucessos, num_experimentos, prob)
psucessos

#4 sinais de quatro tempos; p=? / 0, 1, 2, 3, 4 de sinais verdes

vnum_sucessos = c(0, 1, 2, 3, 4)
vnum_experimentos = 4
vprob = 0.25

vpsucessos = dbinom(vnum_sucessos, vnum_experimentos, vprob)
vpsucessos

# pbinom -> probabiidade cumulativa
num_sucessos = 4  # 0-4
num_experimentos = 4
prob = 0.25

cpsucessos = pbinom(cnum_sucessos, cnum_experimentos, cprob)
cpsucessos

# prova 12 questoes
#Chance de acertar 7, com cad aquestão tendo 4 alternativas

num_sucessos = 7
num_experimentos = 12
prob = 0.25

psucessos = dbinom(num_sucessos, num_experimentos, prob)
psucessos

File: formacao-cientista/2. distribuicoes/dist_binomial.py
from scipy.stats import binom

#AO JOGAR UMA MOEDA 5 VEZES, QUAL A PROBABILIDADE DE SAIR CARA?

num_sucessos = 3
num_experim = 5
prob_sucesso = 0.5

prob = binom.pmf(num_sucessos, num_experim, prob_sucesso)
print(f'Probabilidade 1: {prob}')

#4 sinais de quatro tempos; p=? / 0, 1, 2, 3, 4 de sinais verdes

num_sucessos = [0, 1, 2, 3, 4]
num_experim = 4
prob_sucesso = 0.25

prob = binom.pmf(num_sucessos, num_experim, prob_sucesso)
print(f'Probabilidade 2: {prob}')

cprob = binom.cdf(4, 4, 0.25)
print(f'Probabilidade 3: {cprob}')

# prova 12 questoes
#Chance de acertar 7, com cada aquestão tendo 4 alternativas
prob = binom.pmf(7, 12, 0.25)
print(f'Probabilidade 4: {prob}')

probc = binom.pmf(2, 2, 0.5)
print(f'Probabilidade 5: {probc}')

